src/coreclr/src/vm/threadsuspend.cpp

```diff
@@ -5616,27 +5616,28 @@ void STDCALL OnHijackWorker(HijackArgs * pArgs)
 #endif // HIJACK_NONINTERRUPTIBLE_THREADS
 }
 
-ReturnKind GetReturnKind(Thread *pThread, EECodeInfo *codeInfo)
+bool GetReturnAddressHijackInfo(Thread *pThread, EECodeInfo *codeInfo, void** hijackAddress)
```

Comment:
Nit: I would expect `GetXXX` methods have no side-effects. This method has side-effects - it actually hijacks the  hijackAddress. The `GetXXX` name does not look right for it.

Reply:
Scratch that - I got confused that this method calls `Thread::SetHijackReturnKind`. It does not actually perform the hijack.

Reply:
Still, it would be more appropriate for SetHijackReturnKind to be called by `Thread::HijackThread` - once we actually decide to do the hijack.

Reply:
It doesn't hijack the return address - it just returns address to use for hijacking. But the pThread->SetHijackReturnKind(returnKind); should not be here, which I've not noticed before.